### PR TITLE
mgmt/mcumgr: Split Kconfig into smaller Kconfigs

### PR DESCRIPTION
**Pull request by de-nordic**
The commit splits the main mcumgr Kconfigs into primary Kconfig
that collect general mcumgr options for Zephyr (Zephyr specific
commands, transports, and so on), into lib/Kconfig
that collects library options and command groups' Kconfigs.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

**Review by nvlsianpu — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]

**Comment by carlescufi** [comment]
@de-nordic please rebase

**Review by nvlsianpu — DISMISSED** [review]

**Comment by carlescufi** [comment]
@de-nordic please rebase

**Review by nvlsianpu — DISMISSED** [review]

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/Kconfig [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+config IMG_MGMT_DIRECT_IMAGE_UPLOAD
+	bool "Allow direct image upload"
+	help
+	  Enables directly uploading image to selected image partition.
+	  This changes how "image" is understood by mcumgr: normally the mcumgr allows to upload
+	  to first slot of the only image it knows, where image is understood as two slots
+	  (two DTS images for Zephyr); this allows to treat every DTS defined image as direct
+	  target for upload, and more than two may be used (4 at this time).
+	  NOTE: When direct upload is used the image numbers are shifted by + 1, and the default
+	  behaviour is, when image is not selected, to upload to image that represents secndary
```
Comment (lairdjm) [inline subsys/mgmt/mcumgr/lib/cmd/img_mgmt/Kconfig:56]: *secondary

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: subsys/mgmt/mcumgr/lib/cmd/img_mgmt/Kconfig (hunk 1 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/img_mgmt/Kconfig]: I do not even know hot did that letter disappeared.

**Review by nordicjm — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**